cpp/tests/join/distinct_join_tests.cpp

```diff
@@ -679,3 +680,33 @@ TEST_F(DistinctJoinTest, InvalidLoadFactor)
   // Test load factor > 1
   EXPECT_THROW(cudf::distinct_hash_join(t0, cudf::null_equality::EQUAL, 1.1), cuco::logic_error);
 }
+
+TEST_F(DistinctJoinTest, LargeExtentOverflowPrevention)
+{
+  // Test that validates size_t extent can handle hash table sizes that would
+  // overflow int32_t extent when load_factor < 1.0
+
+  // Use a table size that when divided by low load factor would exceed INT32_MAX
+  // if using int32_t extent, but should work fine with size_t extent
+  constexpr cudf::size_type table_size = 10000000;  // 10M rows
+  constexpr double load_factor         = 0.0025;    // Hash table extent would be ~4B
+
+  // Validate our test assumptions
+  constexpr auto expected_hash_size = static_cast<size_t>(table_size / load_factor);
+  static_assert(expected_hash_size > std::numeric_limits<cudf::size_type>::max(),
+                "Hash table size should be significant");
+
+  auto const init = cudf::numeric_scalar<cudf::size_type>{0};
+  auto build_col  = cudf::sequence(table_size, init, cudf::numeric_scalar<cudf::size_type>{1});
+
+  auto build_table = cudf::table_view{{build_col->view()}};
+  cudf::table empty_probe_table{};
+
+  // This should succeed with size_t extent - would have failed with int32_t extent
+  // in scenarios approaching the overflow boundary
```

Comment:
Same here. I don't want to unnecessarily burden the test machines with these kinds of tests. We can always mark it disabled to keep it from running in CI.